go/adbc/driver/snowflake/statement.go

```diff
@@ -108,6 +108,11 @@ func (st *statement) SetOption(key string, val string) error {
 	case adbc.OptionKeyIngestTargetTable:
 		st.query = ""
 		st.targetTable = val
+	case adbc.OptionKeyIngestTemporary:
+		return adbc.Error{
+			Msg:  fmt.Sprintf("[Snowflake] Unknown statement option '%s'", key),
+			Code: adbc.StatusNotImplemented,
+		}
```

Comment:
Is this really necessary? The default will do this already

Reply:
Ah, I see

Reply:
I think instead of this, the 'default' at the bottom needs to use the NotSupported code instead of InvalidArgument

Reply:
The default https://github.com/apache/arrow-adbc/blob/main/go/adbc/driver/snowflake/statement.go#L145-L149 will throw `StatusInvalidArgument` while https://github.com/apache/arrow-adbc/blob/main/python/adbc_driver_manager/adbc_driver_manager/dbapi.py#L876-L879 checks for `StatusNotImplemented`

If you do not want to change the go implementation, the python manager could pass on ProgrammingError as well,
```
try:
   self._stmt.set_options(**options)
except (NotSupportedError, ProgrammingError):
   pass
```

Reply:
Right, I think the default used here is wrong. The C header specifies NOT_IMPLEMENTED for unrecognized options and Go usually follows this: https://github.com/apache/arrow-adbc/blob/0656c69da28e01f947ba3477fdcd338faf554a85/adbc.h#L1220

Reply:
alternative https://github.com/apache/arrow-adbc/pull/1120 still fixes https://github.com/apache/arrow-adbc/issues/1119 you may close this PR